src/Builder/Pdf/AbstractPdfBuilder.php

```
@@ -124,6 +136,63 @@ public function generate(): GotenbergResponse
         return $pdfResponse;
     }
 
+    public function generateAsync(): string
+    {
+        $operationId = ($this->operationIdGenerator)();
+        $this->logger?->debug('Generating PDF file async with operation id {sensiolabs_gotenberg.operation_id} using {sensiolabs_gotenberg.builder} builder.', [
+            'sensiolabs_gotenberg.operation_id' => $operationId,
+            'sensiolabs_gotenberg.builder' => $this::class,
+        ]);
+
+        $this->webhookExtraHeaders['X-Gotenberg-Operation-Id'] = $operationId;
+        $headers = [
+            'Gotenberg-Webhook-Url' => $this->webhookUrl,
+            'Gotenberg-Webhook-Error-Url' => $this->errorWebhookUrl,
+            'Gotenberg-Webhook-Extra-Http-Headers' => json_encode($this->webhookExtraHeaders, \JSON_THROW_ON_ERROR),
+        ];
+        if (null !== $this->fileName) {
+            $headers['Gotenberg-Output-Filename'] = basename($this->fileName, '.pdf');
+        }
+        $this->gotenbergClient->call($this->getEndpoint(), $this->getMultipartFormData(), $headers);
+
+        return $operationId;
+    }
+
+    public function webhookConfiguration(string $webhook): static
+    {
+        if (null === $this->webhookConfigurationRegistry) {
+            throw new \LogicException('The WebhookConfigurationRegistry is not available.');
```

We should throw an exception from the Sensiolabs\GotenbergBundle namespace